src/hex_tarball.erl

```diff
@@ -43,8 +92,31 @@
 %%        inner_checksum => <<178,12,...>>}}
 %% '''
 %% @end
--spec create(metadata(), files()) -> {ok, {tarball(), checksum()}}.
+-spec create(metadata(), files()) -> {ok, tarball()} | {error, errors(), warnings()}.
 create(Metadata, Files) ->
+    case validate(Metadata, Files) of
+        #{files := _Files1, errors := [], warnings := Warnings} ->
+             Metadata1 = filter_metadata([unknown, deprecated], Metadata, Warnings),
+            {ok, Res} = create_tarball(Metadata1, Files),
+            Final = maps:put(warnings, Warnings, Res),
+            {ok, Final};
+        #{errors := Errors, warnings := Warnings} ->
+            {error, {Errors, Warnings}}
```

Comment:
@ericmj I can not remember why we're returning a hash for `ok` and a tuple for `error`, is this cruft or did we want it this way?